### ISSUE TITLE
Binding the Nuget Package into a UWP-DLL-Project not possible: Namespace not Found

### ISSUE THREAD

**Issue by aatd** [ffmpeginteropx/FFmpegInteropX#239]
Is it possible to make an own c++ dll-project with the Nuget Package? When creating a Blank UWP-Cpp-Project and downloading the Nuget Package everything worked fine.

But now i want to extend a dll for UWP. But when importing this package through the nuget Package Manager it is not recognizing the namespace.  But ir reads the ffmpeg headers

also tried to force reinstalling the Package.

And in all Scenarios i included the UWPFFmpeg build Package to of course!

Best regards!
Asef

**Comment by mcosmin222** [comment]
What do you mean by "But now i want to extend a dll for UWP"? What are you trying to do? Do you want to add new functionality?

**Comment by aatd** [comment]
Hi there thanks for replying. 
So what's the deals:

I'm developing an App for Holographic telepresence with Unity3D. The Device to deploy my Applicaition is the Hololens 1. 
There is a UWP-Mediaplyer Wrapper-Plugin for Unity here https://github.com/vladkol/MediaPlayback.

The Problem now is that I implmented a Server Application hosting the Videostream should deliver Images to my Hololens-App is streaming the Content via RTMP (which as far as I know and expericend is not supported with the Mediaplyer Microsoft is providing).  
I also tried HLS and DASH but is not meeting my requirements compared to RTMP in latency and stability.  

What I'm trying to do now is integrading your Plugin with the one mentioned earlier. 
The Sources for that are available wihtin vladkol's repo here: https://github.com/vladkol/MediaPlayback/tree/master/MediaPlayback.

What I already accomplished was streaming my RTMP stream with the Mediaplayer using your Plugin wihtin the [Blank App (Universal Windows)](https://docs.microsoft.com/de-de/windows/apps/desktop/visual-studio-templates) template deliverd with Visual Studio 2019 (Both C# and C++). For that I just imported your Nuget Packages via the Manager and everything worked just as planned. 

Then I tried the same thing with the "Class Library (Universal Windows)" Template to write an DLL with your Plugin before trying out the real stuff. 

That failed. 

I then tried that integrstion with vladkol's Plugin (In hope it wouldn't end up the same way)

That failed too.

*So what behaviour I expected:*
After installing the Nuget Package having access to all FFmpegInterop namespaces.

*What occured:*
Namespaces not accessable. In Visual Studio though I could see the FFmpegInterop Namespaces within the Object Window.

**Comment by mcosmin222** [comment]
Can you upload the Class Library project somewhere so I can try and reproduce the issue?

As a side note, I personally import the ffmpeginterop cppproj in my own solution, and simply reference it directly. It will still download the ffmpeg build from nuget.

Thanks.

**Comment by aatd** [comment]
[DLL (Universal Windows).zip](https://github.com/ffmpeginteropx/FFmpegInteropX/files/6021191/DLL.Universal.Windows.zip)

Sorry it's name wasn't "Class Library (Universal Windows) (C#-Project)" but "DLL (Windows Universal) (C++-Project)" this one is scaffolded with Visual Studio 2019.

To your Side note:
Did you mean after adding the cppproj to the solution, just make right click on the project **Add** -> **Reference...** or linking the headers with the libs to the Project? I tried the first one and that failed.

And one thing forgot to didn't mention:
On all my attempts I could Include the ffmpeg (like libavformat/** etc.) Headers. Though I don't know if this might help! 

Thanks and best regards!

**Comment by mcosmin222** [comment]
Hmm the problems seems to be that we don't distribute a .h file for you to include.
I think it is better to simply import the ffmpeginterop c++ project in your solution and add the reference directly.

**Comment by lukasf** [comment]
Hi @aatd,

when you reference our NuGet packet, it will automatically include the .winmd file into the project, which contains the definitions and namespaces for the lib. So you do not need to manually include any .h file. But instead of creating a DLL project, you have to create a "Windows Runtime Component (Universal Windows)" project. Then just add our NuGet packet, build once, and you will find our namespaces and classes. I just tried it, it works fine.

**Comment by aatd** [comment]
Hi @lukasf you're right. Though there is a Problem with the requrement. I needed to include the Ffmpeginterop within this dll project: https://github.com/vladkol/MediaPlayback/tree/master/MediaPlayback

But I have found a Solution even for a typical uwp dll project (Though i guess it is a dirty one)

when importing the nuget package within a uwp dll-project it won't find the include files because yours aren't shipping them (these are stored within the winmd file) as @mcosmin222 mentioned.

Enabling the /ZW Flag within the Project Properties fixed this.

Though it might be risky because this flag is forcing the compiler to include these defintion and may cause ambiguities within namespaces. Though I'm not an expert within the UWP Ecosystem.
